cmake/scripts/expand_instantiations.cc

```diff
@@ -52,11 +52,16 @@
 #include <map>
 #include <string>
 
-// a map from the keys in the expansion lists to the list itself. For
+// Returns a map from the keys in the expansion lists to the list itself. For
 // instance, the example above will lead to the entry
-//      expansion_lists[REAL_SCALARS] = (double, float)
+//      get_expansion_lists()[REAL_SCALARS] = (double, float)
 // in this map, among others
-std::map<std::string, std::list<std::string>> expansion_lists;
+std::map<std::string, std::list<std::string>> &
+get_expansion_lists()
```

Comment:
I'm not opposed to this change, but I question the wisdom of the warning. We are here in a user program that depends on libstdc++, and we are constructing an object that only depends on libstdc++ functionality. Since libstdc++ is in a shared lib, all of its global variable constructors run before we even get to `main()`, and consequently by the time we run the constructor of the global variable that was here in the existing code, everything is in fact already constructed and ready to use. There is no ordering question, though it is conceivable that there is an ordering problem with other global variables in this file.

I'm ok with the change in question, primarily because I'm in favor of keeping code uniform. I'm just questioning whether that flag is as useful...

Reply:
Are you fine with the PR if I drop this commit?

Reply:
Let's not worry about it -- it's fine as is!